tests/tracer/test_trace_utils.py

```diff
@@ -33,16 +33,16 @@ def int_config():
 
 
 @pytest.fixture
-def span(tracer):
-    span = tracer.trace(name="myint")
-    yield span
+def span():
+    with tracer.trace(name="myint") as span:
+        yield span
     span.finish()
```

Comment:
since you changed to be a `with` statement, this `span.finish()` is not necessary.... in fact I am not sure it even matters if we call `span.finish()` for any of these tests? since it would be getting called after the test finishes and on cleanup and not called for any assertions?